src/rulesets/packages/NaNMath.jl

```diff
@@ -1,9 +1,11 @@
 module NaNMathGlue
 using ChainRulesCore
 using NaNMath
+using SpecialFunctions
 
 @scalar_rule(NaNMath.sin(x), NaNMath.cos(x))
 @scalar_rule(NaNMath.cos(x), -NaNMath.sin(x))
+@scalar_rule(NaNMath.tan(x), 1 + NaNMath.pow(NaNMath.tan(x), 2))
```

Comment:
This is wrong I think?
Can you cite for this?

Reply:
Wikipedia has the same identity here: [https://en.m.wikipedia.org/wiki/Trigonometric_functions](https://en.m.wikipedia.org/wiki/Trigonometric_functions). But why not just use `1 + Ω^2`? `NaN` should propagate through that just fine.

Reply:
I copied this from DiffRules

Reply:
I think we should change it to `1 + Ω^2`
unless there is a reason not to?